### PR TITLE
Mh customer resource

### PR DESCRIPTION
**Pull request by mhotchkiss84**
# Description
Created the GET, POST, PUT methods for the customers. The user is able to get all customers, get customer by id, post a new customer, and edit customers. 

## Related Ticket(s)
1

## Steps to Test

**Ensure that you have fetched and changed to the mh-customer-resource branch**

**Setup the database**
Step 1:
In Visual Studio create a new query on the BanagzonAPI database.

Step 2:
Copy, paste, and run the following SQL script to populate the database.
[SQL Script](https://gist.github.com/mhotchkiss84/a11e6277bfdd4cace27fc6eedc8646d8)

**Built in tests**
Step 1:
In Visual Studio type test in the search bar and click on "Test Explorer"

Step 2:
Open the dropdown menu on TestBangazonAPI

Step 3:
Locate and select CustomerTest

Step 4:
Click on the single arrow (run) / control+ R, T which with CustomerTest selected will only run the tests in view.

Step 5:
Verify that all customer tests ran and pass (green checks)

**-Testing the GET, POST, PUT methods-**

**--In Visual Studio run without the debugger by hitting control F5--**

**Test get all customers**
Step 1:
Open Postman

Step 2:
In Postman select GET

Step 3:
Paste the following into the address bar in Postman:
https://localhost:44352/api/customers

Step 4:
In Postman click the Send button

Step 5:
Verify that the 4 customers that we inserted into the database are present in the body response from the GET request.

**Test get a single customer by id**
Step 1:
In Postman change the address bar to:
https://localhost:44352/api/customers/3
(3 in the Id of the user that we want to get the information for and can be 1-4 unless you have conflicting ids in your database)

Step 2:
In Postman click on the send button

Step 3:
Verify that customer information comes back without an error. 

**Test create/POST a new customer**
Step 1:
In Postman change the request from a GET to POST

Step 2:
Change the address bar in Postman to:
https://localhost:44352/api/customers

Step 3:
In Postman select body and ensure that JSON is selected

Step 4:
Copy the following JSON and place it into the body section of Postman
{
        "firstName": "Another test",
        "lastName": "Customer",
        "creationDate": "2012-04-23T18:25:43.511Z",
        "lastActiveDate": "2012-04-23T18:25:43.511Z"
}

Step 5:
In Postman click on send

Step 6:
Verify that the response received shows the user with their new Id.

**Test edit/PUT customer**
Step 1:
In Postman change the request to PUT

Step 2:
Change the address bar in Postman to
https://localhost:44352/api/customers/5
(Note that the Id should match the customer we created. If the Id came back as something else in your DB than you need to change the 5 to match the Id of the customer that we are wanted to test changing)

Step 3:
Copy the following JSON and place it into the body in Postman
{
        "firstName": "Test",
        "lastName": "Customer",
        "creationDate": "2012-04-23T18:25:43.511Z",
        "lastActiveDate": "2012-04-23T18:25:43.511Z"
}

Step 4:
In Postman click on Send and it should show up with a 204 No Content.

Step 5:
Change the request in Postman to a GET

Step 6:
Change the address bar in Postman to
https://localhost:44352/api/customers/5

Step 7:
In Postman click Send

Step 8:
Verify in the response that the customer with the Id of 5 has been changed with the JSON we entered on step 3

# Checklist:
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x ] My changes generate no new errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by apreece13 — APPROVED** [review]

**Comment by apreece13** [comment]
Everything worked and looked good to me!

File: .gitignore [hunk 1]
```diff
@@ -42,9 +26,9 @@ x64/
 ipch/
 *.aps
 *.ncb
-*.opensdf
-*.sdf
-*.cachefile
+
+=======
+>>>>>>> 96e5f232be906b4eda4d13ee9b2ecc1ba5ebbb6f
```
Comment (Tommy-Spurlock) [inline .gitignore]: Shouldn't be here. Looks like a _Merge Conflict_ Marker

File: .vs/VSWorkspaceState.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (Tommy-Spurlock) [inline .vs/VSWorkspaceState.json:1]: More Merge Conflict Things. This should be .gitignored

**Review by Tommy-Spurlock — CHANGES_REQUESTED** [review]
Make sure when you name your _Pull Request_ to be more specific. Something like "Users can access the Customers Resource" would be much better. Let me know when you make these changes!

**Comment by Tommy-Spurlock** [comment]
- Add a DROP and CREATE TABLE to SQL script. That way the person testing will have the correct Id's specified.
- Include Directions on how to run the Integrated test.

File: .gitignore (hunk 1 above)
Comment (mhotchkiss84) [reply .gitignore]: Fixed

File: .vs/VSWorkspaceState.json (hunk 2 above)
Comment (mhotchkiss84) [reply .vs/VSWorkspaceState.json:1]: Fixed

**Pull request merged**